File: Java/Sort/src/domain/Book.java
package domain;

public class Book {
	
	private Author author;
	private int numberPages;
	private String title;
	
	public Book(String _title, Author _author, int _numberPages) {
		this.author = _author;
		this.numberPages = _numberPages;
		this.title = _title;
	}
	
	public int compareAuthor(Book _book) {
		return this.author.compareTo(_book.getAuthor());
	}
	
	public int compareNumberPages(Book _book) {
		int _numberPages = _book.getNumberPages();
		if(this.numberPages < _numberPages) {
			return -1;
		} else if(this.numberPages == _numberPages) {
			return 0;
		} else {
			return 1;
		}
	}
	
	public int compareTitle(Book _book) {
		return this.title.compareTo(_book.getTitle());
	}
	
	public Author getAuthor() {
		return this.author;
	}
	
	public int getNumberPages() {
		return this.numberPages;
	}
	
	public String getTitle() {
		return this.title;
	}
}

File: JavaScript/SelectionSort/src/Book.js

  function Book(_title, _author, _numberPages) {

    var self = this;
    self.title = _title;
    self.author = _author;
    self.numberPages = _numberPages;

    function compareAuthor(_book) {
      return self.author.compareTo(_book.getAuthor());
    }

    function compareNumberPages(_book) {
      var compare = 0;
      if(parseInt(self.numberPages) < parseInt(_book.getNumberPages())) {
        compare = -1;
      } else if(parseInt(self.numberPages) > parseInt(_book.getNumberPages())){
        compare = 1;
      }
      return compare;
    }

    function compareTitle(_book) {
      return self.title.localeCompare(_book.getTitle());
    }

    function getAuthor() {
      return self.author;
    }

    function getNumberPages() {
      return self.numberPages;
    }

    function getTitle() {
      return self.title;
    }

    return {
      compareAuthor: compareAuthor,
      compareNumberPages: compareNumberPages,
      compareTitle: compareTitle,
      getAuthor: getAuthor,
      getNumberPages: getNumberPages,
      getTitle: getTitle
    }
  }

  module.exports = Book;
